qutebrowser/browser/webkit/webkittab.py

```diff
@@ -193,159 +196,198 @@ def _on_mode_entered(self, mode):
             #
             # Note: We can't use hasSelection() here, as that's always
             # true in caret mode.
-            if not self.selection_enabled:
+            if self.selection_state is browsertab.SelectionState.NONE:
                 self._widget.page().currentFrame().evaluateJavaScript(
                     utils.read_file('javascript/position_caret.js'))
 
     @pyqtSlot(usertypes.KeyMode)
     def _on_mode_left(self, _mode):
         settings = self._widget.settings()
         if settings.testAttribute(QWebSettings.CaretBrowsingEnabled):
-            if self.selection_enabled and self._widget.hasSelection():
+            if self.selection_state is not \
+                    browsertab.SelectionState.NONE and \
+                    self._widget.hasSelection():
```

Comment:
Please use parentheses around the condition instead, then you won't need to use backslashes for line continuation.